### PR TITLE
rpc: refactor ownership of listener

### PR DESCRIPTION
**Pull request by tamird**
It is now handed down from above (server/server.go).

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3360)

### CODE REVIEW

**Comment by tschottdorf** [comment]
Reviewed 18 of 21 files at r1, 4 of 4 files at r2.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[gossip/client_test.go, line 71 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3360#-K57qDAElh6YrfmEwDfG:-K57qDAElh6YrfmEwDfH:1873811368) ([raw file](https://github.com/cockroachdb/cockroach/blob/7ebb06721425a09688ff0b0e65081eaa228bfea2/gossip/client_test.go#L71)):*
duplicate code here which could be factored outxx, though I don't feel strongly.

---

*[gossip/gossip.go, line 433 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3360#-K57qTm3B4e0UTFfsNLV:-K57qTm3B4e0UTFfsNLW:1338660425) ([raw file](https://github.com/cockroachdb/cockroach/blob/7ebb06721425a09688ff0b0e65081eaa228bfea2/gossip/gossip.go#L433)):*
mind mentioning the `addr` and what it's used for in the comment?

---

*[server/server.go, line 50 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3360#-K57tTxAuiIMrPUFeDz_:-K57tTxAuiIMrPUFeDza:-867275625) ([raw file](https://github.com/cockroachdb/cockroach/blob/7ebb06721425a09688ff0b0e65081eaa228bfea2/server/server.go#L50)):*
call this `util/netutil` instead of the either misleading or manually named import (we already have the same for `util/randutil`).
Also, renaming `{->c}rpc` might be worthwhile? Maybe I'm just too keen on avoiding custom import names.

---

*[util/net/net.go, line 91 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3360#-K57ujCdfs-wqvXxAGzf:-K57ujCdfs-wqvXxAGzg:740859439) ([raw file](https://github.com/cockroachdb/cockroach/blob/7ebb06721425a09688ff0b0e65081eaa228bfea2/util/net/net.go#L91)):*
add an example when that would happen.x

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3360)*

**Comment by tamird** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[gossip/client_test.go, line 71 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3360#-K57qDAElh6YrfmEwDfG:-K57vDQlzpfmfpKj9KUm:920457861) ([raw file](https://github.com/cockroachdb/cockroach/blob/7ebb06721425a09688ff0b0e65081eaa228bfea2/gossip/client_test.go#L71)):*
How do you factor it out? You don't have access to to `t` in the `util` stuff.

---

*[gossip/gossip.go, line 433 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3360#-K57qTm3B4e0UTFfsNLV:-K57x8UjG4hkuN4snKhr:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/7ebb06721425a09688ff0b0e65081eaa228bfea2/gossip/gossip.go#L433)):*
Done.

---

*[server/server.go, line 50 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3360#-K57tTxAuiIMrPUFeDz_:-K57vLqA5W_YiGq3fqAY:-1420543399) ([raw file](https://github.com/cockroachdb/cockroach/blob/7ebb06721425a09688ff0b0e65081eaa228bfea2/server/server.go#L50)):*
Yeah I just put this in `util`. That's where other network-related stuff lives; I don't remember why I decided to put it in a separate package before.

---

*[util/net/net.go, line 91 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3360#-K57ujCdfs-wqvXxAGzf:-K57w_ZhiwQsxI817R1V:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/7ebb06721425a09688ff0b0e65081eaa228bfea2/util/net/net.go#L91)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3360)*

**Comment by tschottdorf** [comment]
:lgtm: 

---

Reviewed 14 of 14 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[gossip/client_test.go, line 71 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3360#-K57qDAElh6YrfmEwDfG:-K57xpZ7AfojPEZlp2jZ:-1512726801) ([raw file](https://github.com/cockroachdb/cockroach/blob/7ebb06721425a09688ff0b0e65081eaa228bfea2/gossip/client_test.go#L71)):*
Not even globally, just in this method. Fine as is as far as I'm concerned.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3360#-:-K58-vMftla0jBcLQTFE:980077423)*

File: rpc/server.go [hunk 1]
```diff
@@ -184,12 +169,14 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	conn, _, err := w.(http.Hijacker).Hijack()
 	if err != nil {
-		log.Infof("rpc hijacking %s: %s", r.RemoteAddr, err)
+		log.Infof("rpc hijacking ", req.RemoteAddr, ": ", err)
```
Comment (bdarnell) [inline rpc/server.go]: This log line is busted.

**Pull request merged**